src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/SemanticRange.cs

```diff
@@ -30,10 +30,8 @@ public int CompareTo(SemanticRange other)
                 return 1;
             }
 
-            Debug.Assert(Range.Start.CompareTo(other.Range.Start) == 0 || !Range.OverlapsWith(other.Range), $"{this} overlapped with {other}");
-
-            // Since overlapping SemanticRanges are STRICTLY FORBIDDEN we need only compare the starts 
-            return Range.Start.CompareTo(other.Range.Start);
+            var startCompare = Range.Start.CompareTo(other.Range.Start);
```

Comment:
For context we need to override CompareTo so that we can sort our array of tokens to properly interleave the Razor and C# (and later HTML) tokens in their correct location.

Reply:
So when things overlap are we guaranteed to have the right outcome? Aka in the interpolated string case what happens from an end users perspective?

Reply:
I don't know that we have any "guarantees" since VS doesn't "officially" support overlapping tokens, but the interpolated string case looks like the following, which I believe is "at intended" for Roslyn.
![image](https://user-images.githubusercontent.com/14987221/104496248-972d2500-558d-11eb-9761-ae1d6b433118.png). If we decide that having the overlaps isn't acceptable because it's not an officially supported VS scenario then Roslyn would have to react, and this change would still be "right", we just would never get past the Start comparison.

Can't check the interpolated string case for VSCode because we don't have SemanticTokens for C# there yet (filed https://github.com/microsoft/vscode/issues/114299 against them now that SemanticTokens are out of preview).

Reply:
Ok cool, I just wanted to make sure we weren't exploding at another layer by returning potentially overlapping tokens to the platform